bundles/org.openhab.binding.mecmeter/src/main/java/org/openhab/binding/mecmeter/handler/MecMeterHandler.java

```diff
@@ -0,0 +1,408 @@
+
+    private @Nullable ScheduledFuture<?> pollFuture;
+
+    private @Nullable MecMeterResponse powerMeterResponse;
+
+    public MecMeterHandler(Thing thing, HttpClient httpClient) {
+        super(thing);
+        this.httpClient = httpClient;
+    }
+
+    @Override
+    public void handleCommand(ChannelUID channelUID, Command command) {
+        if (command instanceof RefreshType) {
+            updateChannel(channelUID.getId());
+        } else {
+            logger.debug("Received unsupported command {}.", command);
+        }
+    }
+
+    /**
+     * function which is called to refresh the data
+     */
+    public void refresh() {
+        updateData();
+        updateChannels();
+    }
+
+    @Override
+    public void dispose() {
+        super.dispose();
+        logger.debug("removing thing..");
+        if (pollFuture != null) {
+            pollFuture.cancel(true);
+        }
+    }
+
+    @Override
+    public void initialize() {
+        MecMeterDeviceConfiguration config = getConfig().as(MecMeterDeviceConfiguration.class);
+        String configCheck = config.isValid();
+
+        if (configCheck != null) {
+            updateStatus(ThingStatus.OFFLINE, ThingStatusDetail.CONFIGURATION_ERROR, configCheck);
+            return;
+        }
+        updateStatus(ThingStatus.UNKNOWN);
+
+        if (pollFuture != null) {
+            pollFuture.cancel(false);
+        }
+        pollFuture = scheduler.scheduleWithFixedDelay(() -> {
+            refresh();
+        }, 0, config.refreshInterval, TimeUnit.SECONDS);
+    }
+
+    /**
+     * Get new data
+     * Function to save Response of the powermeter
+     */
+    private void updateData() {
+        powerMeterResponse = getRealtimeData();
+    }
+
+    /**
+     * Get new realtime data over the network
+     *
+     * @return MecMeterResponse class where json values "are saved"
+     */
+    private @Nullable MecMeterResponse getRealtimeData() {
+        MecMeterResponse result = null;
+        boolean resultOk = false;
+        String errorMsg = null;
+
+        MecMeterDeviceConfiguration config = getConfig().as(MecMeterDeviceConfiguration.class);
+
+        try {
+            String basicAuthentication = "Basic " + Base64.getEncoder()
+                    .encodeToString(new String(USERNAME + ":" + config.password).getBytes(StandardCharsets.ISO_8859_1));
+
+            String location = MecMeterBindingConstants.POWERMETER_DATA_URL.replace("%IP%", config.ip.strip());
+
+            ContentResponse response = httpClient.newRequest(location).method(HttpMethod.GET)
+                    .header(HttpHeader.AUTHORIZATION, basicAuthentication).timeout(API_TIMEOUT, TimeUnit.MILLISECONDS)
+                    .send();
+            if (response.getStatus() != 200) {
+                errorMsg = "Reading meter did not succeed: " + response.getReason();
+                logger.error("Request to meter failed: HTTP {}: {}", response.getStatus(), response.getReason());
```

Comment:
I think this should be warn, if the remote fails.

Reply:
https://www.openhab.org/docs/developer/guidelines.html#f-logging says:

> error logging should only be used
> - in case some code fails irrecoverably and the user should report it as a severe bug.

If we get a response, which is not a HTTP 200 here, I think we have a severe bug here, which should be reported as we seem to be doing invalid requests that make the binding disfunctional.